router/src/server.rs

```diff
@@ -856,25 +856,32 @@ pub async fn run(
         docker_label: option_env!("DOCKER_LABEL"),
     };
 
-    // Create router
-    let app = Router::new()
-        .merge(SwaggerUi::new("/docs").url("/api-doc/openapi.json", ApiDoc::openapi()))
-        // Base routes
+    // Configure Swagger UI
+    let swagger_ui = SwaggerUi::new("/docs").url("/api-doc/openapi.json", ApiDoc::openapi());
+
+    // Define base and health routes
+    let base_routes = Router::new()
         .route("/", post(compat_generate))
         .route("/info", get(get_model_info))
         .route("/generate", post(generate))
         .route("/generate_stream", post(generate_stream))
         .route("/v1/chat/completions", post(chat_completions))
-        // AWS Sagemaker route
-        .route("/invocations", post(compat_generate))
-        // Base Health route
         .route("/health", get(health))
-        // Inference API health route
-        .route("/", get(health))
-        // AWS Sagemaker health route
         .route("/ping", get(health))
-        // Prometheus metrics route
-        .route("/metrics", get(metrics))
+        .route("/metrics", get(metrics));
+
+    // Conditional AWS Sagemaker route
+    let aws_sagemaker_route = if std::env::var("OAI_ENABLED").map_or(false, |val| val == "true") {
```

Comment:
Can you move this into the CLI argument from clap ? Makes truthful values equality. more general (and easier to use for users).

`MESSAGES_API_ENABLED` is maybe a bit more explicit.
Personnally, `CHAT_API_ENABLED` seems a bit more apprioriate. 

Both are fine to me. 
at the very least `OPENAI_ENABLED` would be more explicit than OAI imho.

@OlivierDehaene any preference on the name ?

Reply:
thanks for the quick review, will prefer clap in the next commit 👍

and yea I agree the `CHAT_API_ENABLED` is the most descriptive.... chat implies a conversation where messages do not, but totally open to whatever names best